### PR TITLE
Adding notes about Requests' timeout behavior

### PR DESCRIPTION
**Pull request by Brausepaul**
As discussed in #2228, I added a short note to the quick start and the advanced section about Requests' standard usage of timeouts as this can be unexpected.

### CODE REVIEW

**Comment by Lukasa** [comment]
This change seems to add three names to the AUTHORS file. Why is that?

File: docs/user/advanced.rst [hunk 1]
```diff
@@ -910,7 +910,8 @@ Timeouts
 --------
 
 Most requests to external servers should have a timeout attached, in case the
-server is not responding in a timely manner. Without a timeout, your code may
+server is not responding in a timely manner. By standard, requests do not time
```
Comment (Lukasa) [inline docs/user/advanced.rst]: Please change "By standard"  to "By default".

File: docs/user/quickstart.rst [hunk 2]
```diff
@@ -497,7 +497,8 @@ seconds with the ``timeout`` parameter::
     ``timeout`` is not a time limit on the entire response download;
     rather, an exception is raised if the server has not issued a
     response for ``timeout`` seconds (more precisely, if no bytes have been
-    received on the underlying socket for ``timeout`` seconds).
+    received on the underlying socket for ``timeout`` seconds). If NO timeout is specified explicitly, requests do
```
Comment (Lukasa) [inline docs/user/quickstart.rst]: Please change "NO" to "no".

File: docs/user/advanced.rst (hunk 1 above)
Comment (kennethreitz) [inline docs/user/advanced.rst]: Also, 'Requests' is a proper noun and is to be capitalized :)

**Comment by Brausepaul** [comment]
I made changes according to your comments. My fault about the names, that was me mishandling Git.

**Comment by kennethreitz** [comment]
✨🍰✨

**Pull request merged**